### PR TITLE
Fix coupon API

### PR DESCRIPTION
**Pull request by pimuzzo**

### CODE REVIEW

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/DateWrapper.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.cloudesire.platform.apiclient;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateWrapper
```
Comment (pimuzzo) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/DateWrapper.java]: @gionn @malteo help me to find a better name

Comment (gionn) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/DateWrapper.java]: ISO8601Date

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/ISO8601Date.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public ISO8601Date( Date date )
+    {
+        this.date = date;
+    }
+
+    @Override
+    public String toString()
+    {
+        final SimpleDateFormat dateFormat = new SimpleDateFormat( "yyyy-MM-dd" );
+        return dateFormat.format( date );
```
Comment (malteo) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/ISO8601Date.java]: we can obtain the `DateFormat` from a method specific to every class in hierarchy and avoid the override of the `toString` in every child class

File: cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/CouponApi.java [hunk 3]
```diff
@@ -25,30 +26,44 @@
...
+    @POST( "coupon" )
+    Call<CouponDTO> generate( @Query( "type" ) String type, @Query( "productVersion" ) Integer productVersion,
+            @Query( "product" ) Integer product, @Query( "expirationDate" ) ISO8601DateTime expiration,
+            @Query( "days" ) Integer days, @Query( "plafond" ) BigDecimal plafond );
+
+    @POST( "coupon" )
+    Call<CouponDTO> generate( @Query( "type" ) String type, @Query( "code" ) String code,
+            @Query( "productVersion" ) Integer productVersion, @Query( "product" ) Integer product,
+            @Query( "expiration" ) ISO8601DateTime expiration, @Query( "licenseOnly" ) Boolean licenseOnly,
+            @Query( "value" ) BigDecimal value );
```
Comment (gionn) [inline cloudesire-api-client/src/main/java/com/cloudesire/platform/apiclient/api/CouponApi.java]: try to use the same parameter ordering for all the 3 generate methods (howMany can be the last parameter to differentiate between one or many coupon generation)

there is no need to call generateList, method signature already says that

**Pull request merged**